pytorch_lightning/utilities/fetching.py

```diff
@@ -195,10 +192,11 @@ def __iter__(self) -> Generator[Tuple[Any, bool], None, None]:
         return self.fetching_function()
 
     def reset(self) -> None:
-        self.batches: List = []
-        self.dataloader: Optional[Iterable]
```

Comment:
@tchaton the tests fail if this sets `self.dataloader = None`

What was your intention here?

If you meant to reset it, do we also need to reset `self.stage` and `self.dataloader_iter`?

Reply:
Yes, ti shouldn't reset it. Let me remove it.